src/layers/conv.jl

```diff
@@ -972,35 +971,42 @@ end
 
 function message(egnn::EGNNConv, v_i, v_j, e)
     in_dim = egnn.in_dim
-    h_i = v_i[1:in_dim]
-    h_j = v_j[1:in_dim]
+    h_i = v_i[1:in_dim,:]
+    h_j = v_j[1:in_dim,:]
+
+    N = size(h_i, 2)
 
-    x_i = v_i[in_dim+1:end]
-    x_j = v_j[in_dim+1:end]
+    x_i = v_i[in_dim+1:end,:]
+    x_j = v_j[in_dim+1:end,:]
 
     if isnothing(e)
         a = 1
     else
         a = e[1]
     end
 
-    input = vcat(h_i, h_j, norm(x_i - x_j, 2)^2, a)
+    input = vcat(h_i, h_j, sum((x_i - x_j).^2; dims=1), ones(N)' * a)
```

Comment:
Use `sum(abs2.(x_i - x_j); dims=1)` should have better specialization.

Reply:
Done